notify/notify.go

```diff
@@ -119,109 +143,271 @@ func Now(ctx context.Context) (time.Time, bool) {
 	return v, ok
 }
 
-// A Notifier is a type which notifies about alerts under constraints of the
-// given context.
-type Notifier interface {
-	Notify(context.Context, ...*types.Alert) error
+// A Stage processes alerts constraints under the constraints of the given
+// context.
+type Stage interface {
+	Exec(ctx context.Context, alerts ...*types.Alert) ([]*types.Alert, error)
+}
+
+// Builds a RoutingStage as a map of receivers to Stages
+func BuildRoutingStage(confs []*config.Receiver, tmpl *template.Template, meshWait func() time.Duration, inhibitor *inhibit.Inhibitor, silences *meshprov.Silences, ni *meshprov.NotificationInfos, marker types.Marker) RoutingStage {
+	rs := make(RoutingStage)
+
+	for _, rc := range confs {
+		rs[rc.Name] = createStage(BuildReceiverIntegrations(rc, tmpl), meshWait, inhibitor, silences, ni, marker)
+	}
+
+	return rs
+}
+
+// createStage creates a MultiStage as a pipeline of stages for a receiver
+func createStage(receiverIntegrations []Integration, meshWait func() time.Duration, inhibitor *inhibit.Inhibitor, silences *meshprov.Silences, ni *meshprov.NotificationInfos, marker types.Marker) Stage {
+	var ms MultiStage
+	ms = append(ms, NewLogStage(log.With("step", "inhibit")))
+	ms = append(ms, NewInhibitStage(inhibitor, marker))
+	ms = append(ms, NewLogStage(log.With("step", "silence")))
+	ms = append(ms, NewSilenceStage(silences, marker))
+
+	var fs = make(FanoutStage)
```

Comment:
In Prometheus we standardized to avoid `make` whenever possible. So this would become "FanoutStage{}". Feels a bit ugly at first maybe, but standard > taste :)